airflow/providers/google/cloud/operators/cloud_build.py

```diff
@@ -910,8 +910,9 @@ class BuildProcessor:
 
     def __init__(self, build: Union[Dict, Build]) -> None:
         if isinstance(build, Build):
-            self.build = Build(build)
-        self.build = deepcopy(build)
+            self.build: Union[Dict, Build] = Build(build)
+        else:
+            self.build = deepcopy(build)
```

Comment:
This code looks quite suspecious. Why are we wrapping a Build object in another Build object? How could we then deep-copy it? (The latter par this actually implemented wrong; the previous code deepcopies the input for both Build and dict cases.)

Reply:
Yeah. It is strange. I think the idea was really to make it always "Build" no matter if it was Dict or Build originally. Fixing.

Reply:
Actually - just deepcopy is the only thing that is needed. I think it was some earlier code under if.